src/utils/safe.ts

```diff
@@ -12,6 +12,41 @@ export function getUniqueNonces(transactions: SafeTransactionType[]) {
   ]
 }
 
+// returns subset of given transactions for which a given address has not signed
+export function getUnsignedTxsOfUser({
+  address,
+  transactions,
+}: {
+  address: string
+  transactions: SafeTransactionType[]
+}) {
+  const unsignedTxs: SafeTransactionType[] = []
+  transactions.forEach((transaction: SafeTransactionType) => {
+    let hasSigned = false
+    transaction.confirmations?.forEach((confirmation: GnosisSignature) => {
+      if (confirmation.owner.toLowerCase() === address) {
+        hasSigned = true
+      }
+    })
+    if (!hasSigned) unsignedTxs.push(transaction)
+  })
```

Comment:
Give this a crack (untested)

```suggestion
  const unsignedTxs: SafeTransactionType[] = transactions.filter(tx => {
  return tx.confirmations.some(
    confirmation => confirmation.owner.toLowerCase() === address,
  )
})

```

Reply:
Very nice, just needed a `!` before the `.some` statement